### ISSUE TITLE
"IndexError: list index out of range" using Python 2.7.5

### ISSUE THREAD

**Issue by devmotion** [ajgon/gpg-mailgate.py#1]
I got the following error message (using Python 2.7.5):

```
key = line.split('<')[1].split('>')[0]
IndexError: list index out of range
```

I could reproduce this error on another machine (using Python 2.7.5). If I use Python 3.3.2, I don't get any error message.

As a quick and dirty fix I just added `if ('<' or '>') not in line: continue` in GnuPG/**init.py** before `key = line.split('<')[1].split('>')[0]` is called.

**Comment by ajgon** [comment]
Hmmm... that's interesting, can you give me the output of this two commands please:

```
/usr/bin/gpg --version
/usr/bin/gpg --homedir /home/directory/.gnupg --list-keys --with-colons
```

It appears that you don't have email attached with your public key (which shouldn't be possible in theory at least :smiley: )

**Comment by devmotion** [comment]
The output of the two commands:

```
nobody@mail:/ % /usr/local/bin/gpg --version
gpg (GnuPG) 2.0.21
libgcrypt 1.5.3
Copyright (C) 2013 Free Software Foundation, Inc.
License GPLv3+: GNU GPL version 3 or later <http://gnu.org/licenses/gpl.html>
This is free software: you are free to change and redistribute it.
There is NO WARRANTY, to the extent permitted by law.

Home: ~/.gnupg
Unterstützte Verfahren:
Öff. Schlüssel: RSA, ELG, DSA
Verschlü.: IDEA, 3DES, CAST5, BLOWFISH, AES, AES192, AES256, TWOFISH,
           CAMELLIA128, CAMELLIA192, CAMELLIA256
Hash: MD5, SHA1, RIPEMD160, SHA256, SHA384, SHA512, SHA224
Komprimierung: nicht komprimiert, ZIP, ZLIB, BZIP2
```

```
nobody@mail:/ % /usr/local/bin/gpg --homedir /var/gpg/.gnupg --list-keys --with-colons
tru::1:1377745290:1455578024:3:1:5
pub:u:2048:1:D5C901A4665658B9:1329347624:1455578024::u:::escaESCA:
uid:u::::1329347759::0CB774EA3563BCA7AA5D94B9DA70028B91D533D8::David Widmann <dev@devmotion.de>:
uid:u::::1329347752::5DD205001A7C27611CABF5ECCE9CF9E12489551B::David Widmann <david.widmann@figl-widmann.de>:
uid:u::::1329347738::61254A56017DC0B2C7FE08DF83A40FC7AEA0553B::David Widmann <david@devmotion.de>:
uid:u::::1329347726::16AB2BC095A59938189833147EE194961879F14A::David Widmann <devmotion@devmotion.de>:
uid:u::::1329347692::9B46F8B65BDC97376B93E1B326FFF812E4F38315::David Widmann <david.widmann@devmotion.de>:
uid:u::::1374183070::508603E64C0C2F8F2853D4F2C9B70AAE505E7EF1::David Widmann <david.widmann@mytum.de>:
uid:u::::1377744389::7B64D34ABC6EBA820BB4F24DE3ACE77CBBAAFE43::David Widmann <server-mail@devmotion.de>:
sub:u:2048:1:83A82E48549ED839:1329347624:1455578024:::::esa:

```

As you can see, the second line of the second output starts with 'pub', but contains neither '<' nor '>'. Therefore I added the additional line to GnuPG/**init.py**

I don't know how it's possible that I have no email attached to my public key. I think I created my keys with MacGPG one year ago.

**Comment by ajgon** [comment]
You are also using a gpg 2.0, this can be reason as well, since the format of the output is slightly different. I will install it and do some tests :smile:

Thanks for pointing this out!

**Comment by ajgon** [comment]
I checked the issue more precisely and found out, that gpg2 was indeed an issue. The command used in `GnuPG/__init__.py` when only gpg2 is installed, displays keys in new format with `uid`s.

The truth is, that whole idea of putting `/usr/bin/gpg` there was bad from the start, but as I said, I'm only forking the lib not rewriting it, so I'll leave it as is. That's why any dirty hack is fine with me :), so I implemented your solution, just slightly adjusted the conditional.

Thank you for pointing this out!
